src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -432,53 +455,51 @@ namespace {
 // TODO: Manage ephemeral RCAC/ICAC/NOC storage to avoid a full FabricInfo being needed here.
 FabricInfo gFabricBeingCommissioned;
 
-CHIP_ERROR SendNOCResponse(app::CommandHandler * commandObj, EmberAfNodeOperationalCertStatus status, uint8_t index,
-                           CharSpan debug_text)
+CHIP_ERROR SendNOCResponse(app::CommandHandler * commandObj, const ConcreteCommandPath & path, OperationalCertStatus status,
+                           uint8_t index, const CharSpan & debug_text)
 {
-    app::ConcreteCommandPath path = { emberAfCurrentEndpoint(), OperationalCredentials::Id, Commands::NOCResponse::Id };
-    TLV::TLVWriter * writer       = nullptr;
-
-    VerifyOrReturnError(commandObj != nullptr, CHIP_ERROR_INCORRECT_STATE);
-
-    ReturnErrorOnFailure(commandObj->PrepareCommand(path));
-    writer = commandObj->GetCommandDataIBTLVWriter();
-    ReturnErrorOnFailure(writer->Put(TLV::ContextTag(0), status));
-    if (status == EMBER_ZCL_NODE_OPERATIONAL_CERT_STATUS_SUCCESS)
+    Commands::NOCResponse::Type payload;
+    payload.statusCode = status;
+    if (status == OperationalCertStatus::kSuccess)
     {
-        ReturnErrorOnFailure(writer->Put(TLV::ContextTag(1), index));
+        payload.fabricIndex.Emplace(index);
     }
-    ReturnErrorOnFailure(writer->PutString(TLV::ContextTag(2), debug_text));
-    return commandObj->FinishCommand();
+    if (!debug_text.empty())
+    {
+        payload.debugText.Emplace(debug_text);
```

Comment:
Debug text is max 128. Truncate to 128 if larger

Reply:
Done.